### PR TITLE
update github actions

### PR DESCRIPTION
**Pull request by yucwan**

### CODE REVIEW

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -48,46 +49,83 @@ jobs:
...
+    - name: Deploy api-gateway
+      uses: Azure/spring-cloud-deploy@v1
+      with:
+        azure-subscription: ${{ env.AZURE_SUBSCRIPTION }}
+        action: deploy
+        service-name: ${{ env.SPRING_CLOUD_SERVICE }}
+        app-name: ${{ env.API_GATEWAY }}
+        use-staging-deployment: false
+        package: ${{ github.workspace }}/${{ env.API_GATEWAY_JAR }}
+        # jvm-options: -Xms2048m -Xmx2048m -Dspring.profiles.active=mysql
```
Comment (yucwan) [inline .github/workflows/action.yml:72]: jvm options not working yet due to a bug, will add these lines back before merge.

File: .github/workflows/action.yml [hunk 2]
```diff
@@ -2,20 +2,21 @@ name: Deploy to Azure Spring Cloud with Key Vault
 on: [push]
 
 env:
-  RESOURCE_GROUP: petclinic-demo # customize this
-  SPRING_CLOUD_SERVICE: petclinic-demo # customize this
-  # ==== APPS ====:  
+  AZURE_SUBSCRIPTION: 799c12ba-353c-44a1-883d-84808ebb2216 # customize this
```
Comment (yucwan) [inline .github/workflows/action.yml:5]: will replace ids with placeholders before merge

File: README.md [hunk 3]
```diff
@@ -592,11 +591,11 @@ Add them as secrets to your Key Vault:
...
+    az ad sp create-for-rbac --role contributor --scopes /subscriptions/${SUBSCRIPTION}/resourceGroups/${RESOURCE_GROUP}/providers/Microsoft.KeyVault/vaults/${KEY_VAULT} --sdk-auth
 ```
 Then, follow [the steps here](https://docs.microsoft.com/azure/spring-cloud/spring-cloud-github-actions-key-vault#add-access-policies-for-the-credential) to add access policy for the Service Principal.
 
-In the end, add this service principal as secrets named "AZURE_CREDENTIALS" in your forked GitHub repo following [the steps here](https://docs.microsoft.com/en-us/azure/spring-cloud/spring-cloud-github-actions-key-vault#add-access-policies-for-the-credential).
+In the end, add this service principal as secrets named "AZURE_CREDENTIALS" in your forked GitHub repo following [the steps here](https://docs.microsoft.com/azure/spring-cloud/how-to-github-actions?pivots=programming-language-java#set-up-github-repository-and-authenticate-1).
 
 ### Customize your workflow
 Finally, edit the workfolw file `.github/workflows/action.yml` in your forked repo to fill in the names of resource group and the Azure Spring Cloud instance name that you just created:
```
Comment (selvasingh) [inline README.md:601]: @yucwan  - request you to run spell checker. Also, request you to supply step-by-step instructions for creating an action in GitHub

Comment (selvasingh) [reply README.md:601]: Step-by-step instructions should start from forking the repo, creating a new GitHub Workflow et all.

**Review by selvasingh — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (yucwan) [reply README.md]: Done, when you fork the repo the workflow files are already there so no need to create a new workflow.

**Pull request merged**